corehq/apps/accounting/utils/invoicing.py

```diff
@@ -19,14 +22,33 @@ def _get_all_unpaid_saas_invoices():
     )
 
 
+def get_domains_with_subscription_invoices_overdue(today):
+    invoices = _get_unpaid_saas_invoices_in_downgrade_daterange(today)
+    return _get_domains_over_threshold(invoices, today, get_oldest_overdue_invoice_over_threshold)
+
+
 def _get_unpaid_saas_invoices_in_downgrade_daterange(today):
     return _get_all_unpaid_saas_invoices().filter(
         date_due__lte=today - datetime.timedelta(days=1)
     ).order_by('date_due').select_related('subscription__subscriber')
 
 
-def get_oldest_unpaid_invoice_over_threshold(today, domain):
-    for overdue_invoice in _get_unpaid_saas_invoices_in_downgrade_daterange(today).filter(
+def _get_domains_over_threshold(invoices, today, get_oldest_invoice_fn):
+    for domain in set(invoices.values_list(
+        'subscription__subscriber__domain', flat=True
+    )):
+        overdue_invoice, total_overdue_to_date = get_oldest_invoice_fn(today, domain)
+        if overdue_invoice:
+            yield domain, overdue_invoice, total_overdue_to_date
+
+
+def get_oldest_overdue_invoice_over_threshold(today, domain):
+    invoices = _get_unpaid_saas_invoices_in_downgrade_daterange(today)
+    return _get_oldest_invoice_over_threshold(domain, invoices)
```

Comment:
This is actually `_get_oldest_overdue_invoice_over_threshold`. I wonder does a private function adds value here. The function `get_oldest_overdue_invoice_over_threshold` seems like a wrapper of private function, and it makes the name of both functions hard to differentiate both functions.